webapp/blog/views.py

```diff
@@ -1,225 +1,32 @@
 import flask
-import webapp.api.blog as api
-from webapp.api.exceptions import ApiCircuitBreaker, ApiError
-from webapp.blog import logic
 
-blog = flask.Blueprint(
-    "blog", __name__, template_folder="/templates", static_folder="/static"
-)
+from canonicalwebteam.blog import wordpress_api
+from canonicalwebteam.blog import logic
+from canonicalwebteam.blog.app import BlogExtension
+from webapp.api.exceptions import ApiError
 
 
-@blog.route("/")
-def homepage():
-    BLOG_CATEGORIES_ENABLED = (
-        flask.current_app.config["BLOG_CATEGORIES_ENABLED"] == "true"
-    )
-
-    page_param = flask.request.args.get("page", default=1, type=int)
-
-    # Feature flag
-    if BLOG_CATEGORIES_ENABLED:
-        filter = flask.request.args.get("filter", default=None, type=str)
-
-        if filter == "all":
-            filter = None
+def init_blog(app, url_prefix):
+    blog = BlogExtension()
+    blog.init_app(app, "Snapcraft Blog", [2996], "", url_prefix)
 
+    @app.route(url_prefix + "/api/snap-posts/<snap>")
+    def snap_posts(snap):
         try:
-            categories_list = api.get_categories()
+            blog_tags = wordpress_api.get_tag_by_name(
+                "".join(["sc:snap:", snap])
+            )
         except ApiError:
```

Comment:
I think `ApiError` won't work here, since `wordpress_api` does not raise this exception.

Reply:
Same